src/core/thread/mle.cpp

```diff
@@ -1185,6 +1198,27 @@ void Mle::HandleParentRequestTimer(void)
     }
 }
 
+void Mle::HandleChildDatasetSyncTimer(void *aContext)
+{
+    static_cast<Mle *>(aContext)->HandleChildDatasetSyncTimer();
+}
+
+void Mle::HandleChildDatasetSyncTimer(void)
```

Comment:
Using a timer to poll the state will work, but I think we can do better.

1. In `Mle::HandleAnnounce`, we should also set `mRetrieveNewNetworkData = true`, that will cause the device to retrieve the full Active/Pending Operational Datasets.
2. In `Mle::HandleDataResponse`, if `mPreviousPanId` is set, we send the Announce messages.
3. In `Mle::SetStateChild`, we no longer reset `mPreviousPanId`, because this will be cleared in step 2 above.

Thoughts?